File: 204-count-primes/204-count-primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        vector<int> dp(n,1);
        if (n<=1) return 0;
        int count = 0;
        for(int i =2;i<n;i++){
            if (dp[i]==1){
                count ++;
                for(int j = 2*i;j<n;j=i+j) dp[j] = 0;
            }
        }
        return(count);
    }
};

File: 204-count-primes/204-count-primes.py
class Solution:
    def countPrimes(self, n: int) -> int:
        dp = [True for _ in range(n)];count = 0;
        if n <=1: return 0
        i = 2
        while(i*i<n):
            if dp[i]:
                for j in range(i*i,n,i):
                    dp[j] = False;
            i += 1
        return sum(dp)-2;
                
        